External/FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp

```diff
@@ -775,17 +775,54 @@ DEF_OP(VFAdd) {
 }
 
 DEF_OP(VFAddP) {
-  auto Op = IROp->C<IR::IROp_VFAddP>();
-  switch (Op->Header.ElementSize) {
-    case 4: {
-      faddp(GetDst(Node).V4S(), GetSrc(Op->VectorLower.ID()).V4S(), GetSrc(Op->VectorUpper.ID()).V4S());
-    break;
+  const auto Op = IROp->C<IR::IROp_VFAddP>();
+  const auto OpSize = IROp->Size;
+
+  const auto ElementSize = Op->Header.ElementSize;
+
+  const auto Dst = GetDst(Node);
+  const auto VectorLower = GetSrc(Op->VectorLower.ID());
+  const auto VectorUpper = GetSrc(Op->VectorUpper.ID());
+
+  if (HostSupportsSVE) {
+    const auto Pred = OpSize == 32 ? PRED_TMP_32B.Merging()
+                                   : PRED_TMP_16B.Merging();
+
+    switch (ElementSize) {
+      case 2: {
+        faddp(Dst.Z().VnH(), Pred, VectorLower.Z().VnH(), VectorUpper.Z().VnH());
+        break;
+      }
+      case 4: {
+        faddp(Dst.Z().VnS(), Pred, VectorLower.Z().VnS(), VectorUpper.Z().VnS());
```

Comment:
Looks like we have an issue here where behaviour changes between ASIMD and SVE faddp.
Destination must equal vn since there is only enough encoding space for two registers, causing us to destructively write to VectorLower and also won't actually write to the Dst here.
Need a temporary and moves here for this.